src/util/rop_rad.jl

```diff
@@ -0,0 +1,276 @@
+
+""
+function run_rad(data, model_constructor, optimizer;
+    time_limit::Float64=3600.0,
+    averaging_window::Int = 100,
+    partition_min::Int = 2,
+    partition_max::Int = 5,
+    iteration_with_no_improvement_limit::Int=10,
+    fail_to_improve_limit::Float64=0.8,
+    fail_time_limit::Float64=0.8,
+    rng=Random.GLOBAl_RNG,
+    kwargs...
+    )
+
+    ## create stats
+    solution = Dict{String,Any}(
+    "objective_lb" => 0.0,
+    "objective" => 0.0,
+    "solution" => Dict{String,Any}(),
+    "solve_time" => 0.0
+    )
+    stats = Dict{String,Any}(
+    "ENS" => Dict{Int,Vector{Float64}}(),
+    "sub_ENS" => Dict{Int,Vector{Float64}}(),
+    "improvement" => Dict{Int,Vector{Float64}}(),
+    "solve_time" => Dict{Int,Vector{Float64}}(),
+    "termination_status" => Dict{Int,Vector{_PM.TerminationStatusCode}}(),
+    "primal_status" => Dict{Int,Vector{_PM.ResultStatusCode}}(),
+    "average_fail_to_improve"=>Dict{Int,Vector{Float64}}(),
+    "average_termination_time_limit"=>Dict{Int,Vector{Float64}}(),
+    "solver_time_limit"=>Dict{Int,Vector{Float64}}(),
+    "partition_max"=>Dict{Int,Vector{Float64}}(),
+    "partition_size"=>Dict{Int,Vector{Float64}}(),
+    "repair_count"=>Dict{Int,Vector{Float64}}(),
+    "feasible_period"=>Dict{Int,Bool}(),
+    )
+
+
+    ## Start algorithm
+    t_start = time()
+    network_count = count_repairable_components(data)
+
+    ## initial ordering (utilization heuristic)
+    repair_ordering = utilization_repair_order(data)
+    ens_dict = Dict(k=>sum(load["pd"] for (id,load) in data["load"]) for k in keys(repair_ordering))
+
+    solution["solution"] = replicate_restoration_network(data,count=length(keys(repair_ordering)))
+    solution["solution"] = apply_restoration_sequence!(solution["solution"],repair_ordering)
+    for (nwid,net) in solution["solution"]["nw"] #initialize load solution to 0
+        for (loadid,load) in net["load"]
+            load["pd"] = 0.0
+        end
+    end
+    delete!(solution["solution"]["nw"],"0")
```

Comment:
Wondering about this "0" index network, to discuss.

Reply:
Discussed offline, result of ROP problem starting with a no-repair period.